slack_cli_hooks/hooks/doctor.py

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python
+import json
+import platform
+
+from slack_cli_hooks.protocol import (
+    Protocol,
+    build_protocol,
+)
+
+PROTOCOL: Protocol
+
+
+doctor_payload = {
+    "versions": [
+        {
+            "name": "python",
+            "current": platform.python_build()[0],
+        },
+        {
+            "name": "implementation",
+            "current": platform.python_implementation(),
+        },
+        {
+            "name": "compiler",
+            "current": platform.python_compiler(),
```

Comment:
The [python_compiler()](https://docs.python.org/3.9/library/platform.html#platform.python_compiler) is said to identify the compiler used for compiling Python. I don't think this will be very informative, unless a developer is compiling their own version of python (if they are, not sure how we could help them)

Providing the system information with [platform.system()](https://docs.python.org/3.9/library/platform.html#platform.system) may be more informative for maintainers try to and help a developer, let me know what you think

Reply:
Open to removing this if you don't think it'd be useful. I was honestly looking to make a group of three and the compiler seemed interesting 👀

The system OS and architecture is also included a bit early in the `doctor` output and I think would be the same for the `platform.system()`? I really hope at least 😳 So this base might be covered already 🙌

Reply:
> The system OS and architecture is also included a bit early in the doctor output and I think would be the same for the platform.system()? I really hope at least 😳 So this base might be covered already 🙌

Just saw it now 👍 alternatively `platform.platform()` yields a string similar to this on my machine `macOS-14.4-arm64-arm-64bit` if this information if not outputted by the CLI already I think it would be more useful then [python_compiler()](https://docs.python.org/3.9/library/platform.html#platform.python_compiler)

But if it is [python_compiler()](https://docs.python.org/3.9/library/platform.html#platform.python_compiler) can fill the 3rd item

Reply:
I think the GOOS / GOARCH covers the platform well enough for debugging purposes! Going to keep the compiler here for the time being